types/roslib/index.d.ts

```diff
@@ -77,173 +78,258 @@ declare namespace ROSLIB {
-         *   * error - the error message reported by ROS
+         * @param {string} topicType - The topic type to find.
+         * @param {function} callback - Function with the following params:
+         * @param {string[]} callback.topics - Array of topic names.
+         * @param {function} [failedCallback] - The callback function when the service call failed with params:
+         * @param {string} failedCallback.error - The error message reported by ROS.
          */
         getTopicsForType(
             topicType: string,
             callback: (topics: string[]) => void,
-            failedCallback?: (error: any) => void,
+            failedCallback?: (error: string) => void,
         ): void;
 
         /**
-         * Retrieves list of active service names in ROS.
+         * Retrieve a list of active service names in ROS.
          *
-         * @param callback - function with the following params:
-         *   * services - array of service names
-         * @param failedCallback - the callback function when the ros call failed (optional). Params:
-         *   * error - the error message reported by ROS
+         * @param {function} callback - Function with the following params:
+         * @param {string[]} callback.services - Array of service names.
+         * @param {function} [failedCallback] - The callback function when the service call failed with params:
+         * @param {string} failedCallback.error - The error message reported by ROS.
          */
-        getServices(callback: (services: string[]) => void, failedCallback?: (error: any) => void): void;
+        getServices(callback: (services: string[]) => void, failedCallback?: (error: string) => void): void;
 
         /**
-         * Retrieves list of services in ROS as an array as specific type
+         * Retrieve a list of services in ROS as an array as specific type.
          *
-         * @param serviceType service type to find:
-         * @param callback function with params:
-         *   * topics - Array of service names
-         * @param failedCallback - the callback function when the ros call failed (optional). Params:
-         *   * error - the error message reported by ROS
+         * @param {string} serviceType - The service type to find.
+         * @param {function} callback - Function with the following params:
+         * @param {string[]} callback.topics - Array of service names.
+         * @param {function} [failedCallback] - The callback function when the service call failed with params:
+         * @param {string} failedCallback.error - The error message reported by ROS.
          */
         getServicesForType(
             serviceType: string,
-            callback: (services: string[]) => void,
-            failedCallback?: (error: any) => void,
+            callback: (topics: string[]) => void,
+            failedCallback?: (error: string) => void,
         ): void;
 
```

Comment:
Move `getServiceRequestDetails` and `getServiceResponseDetails` definitions here, to match the source code.